packages/api/tests/lib/rules/test-createRuleResources.js

```diff
@@ -0,0 +1,358 @@
+  const kinesisRule = fakeRuleRecordFactory({
+    type: 'kinesis',
+    workflow,
+    enabled: true,
+    value: randomString(),
+  });
+  // create rule
+  const createdRule = await createRuleTrigger(kinesisRule);
+  const kinesisEventMappings = await getKinesisEventMappings();
+  const consumerEventMappings = kinesisEventMappings[0].EventSourceMappings;
+  const logEventMappings = kinesisEventMappings[1].EventSourceMappings;
+
+  t.is(consumerEventMappings.length, 1);
+  t.is(logEventMappings.length, 1);
+  t.is(consumerEventMappings[0].UUID, createdRule.arn);
+  t.is(logEventMappings[0].UUID, createdRule.log_event_arn);
+
+  t.is(createdRule.name, kinesisRule.name);
+  t.is(createdRule.value, kinesisRule.value);
+  t.false(createdRule.arn === undefined);
+  t.false(createdRule.log_event_arn === undefined);
+
+  // Clean Up
+  // await deleteRuleResources(t.context.testKnex, kinesisRule);
+  await deleteKinesisEventSourceMappings();
+});
+
+test('Creating an invalid kinesis type rule does not add event mappings', async (t) => {
+  const kinesisRule = fakeRuleRecordFactory({
+    type: 'kinesis',
+    workflow,
+    enabled: true,
+    value: randomString(),
+  });
+  delete kinesisRule.name;
+  await t.throwsAsync(createRuleTrigger(kinesisRule), { name: 'SchemaValidationError' });
+
+  const kinesisEventMappings = await getKinesisEventMappings();
+  const consumerEventMappings = kinesisEventMappings[0].EventSourceMappings;
+  const logEventMappings = kinesisEventMappings[1].EventSourceMappings;
+
+  console.log(JSON.stringify(kinesisEventMappings));
+
+  t.is(consumerEventMappings.length, 0);
+  t.is(logEventMappings.length, 0);
+});
+
+test('Creating a rule trigger SQS rule fails if queue does not exist', async (t) => {
+  const rule = fakeRuleRecordFactory({
+    workflow,
+    type: 'sqs',
+    value: 'non-existent-queue',
+    enabled: true,
+  });
+  await t.throwsAsync(
+    createRuleTrigger(rule),
+    { message: /SQS queue non-existent-queue does not exist/ }
+  );
+});
+
+test('Creating a rule trigger for an SQS rule fails if there is no redrive policy on the queue', async (t) => {
+  const queueUrl = await SQS.createQueue(randomId('queue'));
+  const rule = fakeRuleRecordFactory({
+    workflow,
+    type: 'sqs',
+    value: queueUrl,
+    enabled: true,
+  });
+  await t.throwsAsync(
+    createRuleTrigger(rule),
+    { message: `SQS queue ${queueUrl} does not have a dead-letter queue configured` }
+  );
+  t.teardown(async () => await SQS.deleteQueue(queueUrl));
+});
+
+test('Creating a rule trigger for an SQS rule succeeds', async (t) => {
+  const queues = await createSqsQueues(randomString());
+  const rule = fakeRuleRecordFactory({
+    workflow,
+    type: 'sqs',
+    value: queues.queueUrl,
+    enabled: true,
+    meta: {
+      visibilityTimeout: 100,
+      retries: 4,
+    },
+  });
+  const sqsRule = await createRuleTrigger(rule);
+  t.deepEqual(sqsRule, rule);
```

Comment:
I don't know if this is proving anything? Since you're including the `meta` in the `rule` fixture, there's no way to know whether that property was added by the `validateAndUpdateSqsRule` function. I would remove `meta` from the `rule` fixture then compare `sqsRule` against a rule with the expected `meta`

Reply:
I see that the test below proves that the expected `meta` gets set. So is this test necessary?

Reply:
Hmm the first test uses values for meta fields that aren't default. But I can see your perspective

Reply:
Fair enough. I can see that now. Thanks for clarifying